sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlignUpdateAssignmentsSuite.scala

```diff
@@ -0,0 +1,781 @@
+                fail(s"Unexpected children for 's.n_s': $nsValueChildren")
+            }
+
+          case sValueChildren =>
+            fail(s"Unexpected children for 's': $sValueChildren")
+        }
+
+        assert(txt.name == "txt" && txt == txtValue)
+
+      case assignments =>
+        fail(s"Unexpected assignments: $assignments")
+    }
+
+    val sql2 = "UPDATE nested_struct_table SET s.n_s = named_struct('dn_i', 1, 'dn_l', 1L)"
+    parseAndAlignAssignments(sql2) match {
+      case Seq(
+          Assignment(i: AttributeReference, iValue: AttributeReference),
+          Assignment(s: AttributeReference, sValue: CreateNamedStruct),
+          Assignment(txt: AttributeReference, txtValue: AttributeReference)) =>
+
+        assert(i.name == "i" && i == iValue)
+
+        assert(s.name == "s")
+        sValue.children match {
+          case Seq(
+              StringLiteral("n_i"), GetStructField(_, _, Some("n_i")),
+              StringLiteral("n_s"), nsValue: CreateNamedStruct) =>
+
+            nsValue.children match {
+              case Seq(
+                  StringLiteral("dn_i"), IntegerLiteral(1),
+                  StringLiteral("dn_l"), LongLiteral(1L)) =>
+                // OK
+
+              case nsValueChildren =>
+                fail(s"Unexpected children for 's.n_s': $nsValueChildren")
+            }
+
+          case sValueChildren =>
+            fail(s"Unexpected children for 's': $sValueChildren")
+        }
+
+        assert(txt.name == "txt" && txt == txtValue)
+
+      case assignments =>
+        fail(s"Unexpected assignments: $assignments")
+    }
+
+    val sql3 = "UPDATE nested_struct_table SET s.n_s = named_struct('dn_l', 1L, 'dn_i', 1)"
+    parseAndAlignAssignments(sql3) match {
+      case Seq(
+          Assignment(i: AttributeReference, iValue: AttributeReference),
+          Assignment(s: AttributeReference, sValue: CreateNamedStruct),
+          Assignment(txt: AttributeReference, txtValue: AttributeReference)) =>
+
+        assert(i.name == "i" && i == iValue)
+
+        assert(s.name == "s")
+        sValue.children match {
+          case Seq(
+              StringLiteral("n_i"), GetStructField(_, _, Some("n_i")),
+              StringLiteral("n_s"), nsValue: CreateNamedStruct) =>
+
+            nsValue.children match {
+              case Seq(
+                  StringLiteral("dn_i"), GetStructField(_, _, Some("dn_i")),
+                  StringLiteral("dn_l"), GetStructField(_, _, Some("dn_l"))) =>
+                // OK
+
+              case nsValueChildren =>
+                fail(s"Unexpected children for 's.n_s': $nsValueChildren")
+            }
+
+          case sValueChildren =>
+            fail(s"Unexpected children for 's': $sValueChildren")
+        }
+
+        assert(txt.name == "txt" && txt == txtValue)
+
+      case assignments =>
+        fail(s"Unexpected assignments: $assignments")
+    }
+
+    val sql4 = "UPDATE nested_struct_table SET s.n_i = 1"
```

Comment:
@johanl-db, here is the test we talked about. If you have time to contribute any other tests or to check the alignment logic works for Delta, it would be great!